physics/SFC_Models/Land/Noahmp/lnd_iau_mod.F90

```diff
@@ -0,0 +1,848 @@
+      integer, intent(in):: ncid
+      character(len=*), intent(in)::  dim_name
+      integer, intent(out):: dim_len
+      integer :: dimid
+      integer :: errflg
+      character(len=*) :: errmsg_out
+      integer :: status
+
+      !Errors messages handled through CCPP error handling variables
+      errmsg_out = ''
+      errflg = 0
+
+      status = nf90_inq_dimid(ncid, dim_name, dimid)
+      CALL netcdf_err(status, 'reading dim id '//trim(dim_name), errflg, errmsg_out)
+      if (errflg .ne. 0) return
+      status = nf90_inquire_dimension(ncid, dimid, len = dim_len)
+      CALL netcdf_err(status, 'reading dim length '//trim(dim_name), errflg, errmsg_out)
+
+   end subroutine get_nc_dimlen
+
+   subroutine get_var1d(ncid, dim_len, var_name, var_arr, errflg, errmsg_out)
+      integer, intent(in):: ncid, dim_len
+      character(len=*), intent(in)::  var_name
+      real(kind=kind_phys), intent(out):: var_arr(dim_len)
+      integer :: errflg
+      character(len=*) :: errmsg_out
+      integer :: varid, status
+
+      !Errors messages handled through CCPP error handling variables
+      errmsg_out = ''
+      errflg = 0
+
+      status = nf90_inq_varid(ncid, trim(var_name), varid)
+      CALL NETCDF_ERR(status, 'getting varid: '//trim(var_name), errflg, errmsg_out)
+      if (errflg .ne. 0) return
+      status = nf90_get_var(ncid, varid, var_arr)
+                  ! start = (/1/), count = (/dim_len/))
+      CALL NETCDF_ERR(status, 'reading var: '//trim(var_name), errflg, errmsg_out)
+
+   end subroutine get_var1d
+
+   subroutine get_var3d_values(ncid, varid, is,ix, js,jy, ks,kz, var3d, status)
+      integer, intent(in):: ncid, varid
+      integer, intent(in):: is, ix, js, jy, ks,kz
+      real(kind=kind_phys), intent(out):: var3d(ix, jy, kz)   !var3d(is:ie,js:je,ks:ke)
+      integer, intent(out):: status 
+      ! integer, dimension(3):: start, nreco
+      ! start(1) = is; start(2) = js; start(3) = ks
+      ! nreco(1) = ie - is + 1
+      ! nreco(2) = je - js + 1
+      ! nreco(3) = ke - ks + 1
+
+      status = nf90_get_var(ncid, varid, var3d, &  !start = start, count = nreco)
+               start = (/is, js, ks/), count = (/ix, jy, kz/))
+               ! start = (/is, js, ks/), count = (/ie - is + 1, je - js + 1, ke - ks + 1/))
+
+   end subroutine get_var3d_values
+
+   subroutine get_var3d_values_int(ncid, varid, is,ix, js,jy, ks,kz, var3d, status)
+      integer, intent(in):: ncid, varid
+      integer, intent(in):: is, ix, js, jy, ks,kz
+      integer, intent(out):: var3d(ix, jy, kz)   !var3d(is:ie,js:je,ks:ke)
+      integer, intent(out):: status 
+
+      status = nf90_get_var(ncid, varid, var3d, &  !start = start, count = nreco)
+               start = (/is, js, ks/), count = (/ix, jy, kz/))
+               ! start = (/is, js, ks/), count = (/ie - is + 1, je - js + 1, ke - ks + 1/))
+
+   end subroutine get_var3d_values_int
+  
+end module land_iau_mod

```

Comment:
Same here, missed the error handling?

Reply:
ditto, error handling done by netcdf_err